lib/src/components/NxToast/__tests__/NxToastContainer.test.tsx

```diff
@@ -0,0 +1,161 @@
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React, { useState } from 'react';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
+import { screen, render, fireEvent } from '@testing-library/react';
+
+import userEvent from '@testing-library/user-event';
+
+import {reject, propEq } from 'ramda';
+
+import useToggle from '../../../util/useToggle';
+
+import NxToastContainer from '../NxToastContainer';
+import { NxToastContainerProps } from '../types';
+import NxToast from '../NxToast';
+import { NxErrorAlert } from '../../NxAlert/NxAlert';
+
+describe('NxToastContainer', function() {
+  const simpleProps: NxToastContainerProps = {
+    children: []
+  };
+
+  const quickRender = rtlRender<NxToastContainerProps>(NxToastContainer, simpleProps),
+      renderEl = rtlRenderElement(NxToastContainer, simpleProps);
+
+  it('renders a <div> with a class of nx-toast__wrapper', function() {
+    const component = quickRender();
+    const toastWrapper = component.container.children[0];
+
+    expect(toastWrapper.nodeName).toEqual('DIV');
+    expect(toastWrapper).toHaveClass('nx-toast__wrapper');
+  });
+
+  it('contains a<div> with a class of nx-toast__container', function() {
+    const component = renderEl();
+    const toastContainer = component?.querySelector('.nx-toast__container');
+
+    expect(toastContainer).toBeInTheDocument();
+    expect(component).toContainElement(toastContainer as HTMLElement);
+  });
+
+  it('tests NxToast mounting inside NxToastContainer, and unmounting when close button is clicked',
+      async function() {
+
+        const user = userEvent.setup();
+
+        const MountingBehavior = () => {
+          const [showToast, toggleToast] = useToggle(false);
+          return (
+            <>
+              <button type="button" onClick={()=>toggleToast()}>Show Toast</button>
+              <NxToastContainer>
+                {showToast &&
+                <NxToast onClose={()=>toggleToast()}>
+                  <NxErrorAlert>message</NxErrorAlert>
+                </NxToast>
+                }
+              </NxToastContainer>
+            </>
+          );
+        };
+        const { container } = render(<MountingBehavior/>);
+
+        expect(screen.queryByRole('alert')).not.toBeInTheDocument();
+
+        const button = screen.getByRole('button', {name: 'Show Toast'});
+        await user.click(button);
+
+        expect(screen.getByRole('alert')).toBeVisible();
+
+        const toast = container.querySelector('.nx-toast');
+        const closeBtn = screen.getByRole('button', {name: 'Close'});
+        await user.click(closeBtn);
+        fireEvent.animationEnd(toast as Element);
+
+        expect(toast).not.toBeInTheDocument();
+      });
+
+  it('tests focusing when NxToasts mount and unmount', async function() {
+    const user = userEvent.setup();
+
+    interface ToastModel {
+      id: number;
+    }
+
+    const FocusBehavior = () => {
```

Comment:
Why does the test require this ad-hoc component?

Reply:
> Why does the test require this ad-hoc component?

My thought was that I needed to test for the focus as toasts were added and removed from the DOM, and therefore needed functions to call to do it (which I may be assuming incorrectly). Should I render an NxContainer with several NxToasts already (with the { onClose = ()=>{}} prop applied), and test the focus as the onClose function is called? However, I know I also need to test the focus after all toasts are removed, so perhaps a separate test with a component that contains an additional focusable element?

Reply:
> Should I render an NxContainer with several NxToasts already (with the { onClose = ()=>{}} prop applied), and test the focus as the onClose function is called?

Yes that's what I'd do.  One test for when there are other remaining toasts (to see that the last other toast gets focused), another test that's basically the same but with the closing toast being the last one (to see that the last one _besides_ this one gains focus), and another test for when the closing toast is the only toast, to see that focus goes to the other recently focused element.

I'm not sure that you'll need it for these tests, but be aware that RTL has a `rerender` function for when you want to change the props or children of a rendered component.  You don't have to wrap it in another component with internal logic and interactive UI to effect those prop changes.